### ISSUE TITLE
Creating duplicate Messageboard Groups

### ISSUE THREAD

**Issue by lawso017** [thredded/thredded#318]
Creating a new Messageboard Group does not give any visual feedback, so I ended up doing it a few times and then creating duplicates with the same name. Would you be open to a PR that:

1) Provided a success message upon new Messageboard Group creation; and
2) Validates_uniqueness_of a Messageboard Group by name?

Finally, I may be missing this, but is there a recommended way to control the sort order in which Messageboard Groups and Messageboards are presented to the user?

**Comment by glebm** [comment]
Both 1) and 2) would be great to have!
Unfortunately, there is no way to control the sort order yet. Within a group, the messageboards are ordered in the order they were created at. Groups themselves are also ordered in the order they were created at.

**Comment by jayroh** [comment]
The ordering is one of those things where a corner of some sort of
administration would be great to manage this stuff. An order column for
both Messageboard Group and Messageboard would be a good thing to shoot for
in the future.

**Comment by TheDudeWithTheThing** [comment]
doing it

**Comment by Nerian** [comment]
I am actually interesting in having Messageboard Groups with the same name. So a validation on the uniqueness is a bad thing for me. I am interesting in having Messageboard with the ame name too.

I am developing a multi-tenant application. So each tenant have a forum. There is a MessageBoardGroup for each tenant, named after the tenant – so they may have the same name. And inside the MessageboardGroup there should be two boards created by default: 'Admin' and 'Members'. So the unique validation on the Messageboard is also a problem for me.

Is there a way to remove or bypass that validation?

**Comment by glebm** [comment]
@Nerian Perhaps the best would be to have your app modify the uniqueness validation options to include the `scope` option, changing the uniqueness requirement to be "unique within the tenant". This can be done via the public `validators_on` API:

```ruby
Thredded::MessageboardGroup.validators_on(:name)[0].options[:scope] = :tenant_id
```

Alternatively, you can completely remove the uniqueness validation using the private `_validators` and `_validate_callbacks` attributes, but this is gnarly:

```ruby
Thredded::MessageboardGroup._validators[:name].delete_if do |v|
  v.is_a? ActiveRecord::Validations::UniquenessValidator
end
Thredded::MessageboardGroup._validate_callbacks.each do |c|
  if c.raw_filter.is_a ActiveRecord::Validations::UniquenessValidator
    c.raw_filter.attributes.delete :name
  end
end
```

**Comment by glebm** [comment]
Refs PR #345

**Comment by glebm** [comment]
Resolved by #345.

@Nerian You'll also need to change or remove the new index.
